samples/nrf9160/azure_iot_hub/src/main.c

```diff
@@ -527,16 +531,29 @@ static int dps_run(struct azure_iot_hub_buf *hostname, struct azure_iot_hub_buf
 void main(void)
 {
 	int err;
+
+#if IS_ENABLED(CONFIG_DEVICE_ID_USE_IMEI)
+	char device_id[HW_ID_LEN];
```

Comment:
I think we should keep the existing size of the buffer, as the device may be assigned a larger device ID during DPS and that needs to fit inside the buffer.